core/common/src/-CommonPlatform.kt

```diff
@@ -21,30 +21,18 @@
 
 package kotlinx.io
 
-internal expect fun ByteArray.toUtf8String(): String
-
 internal expect fun String.asUtf8ToByteArray(): ByteArray
 
-// TODO make internal https://youtrack.jetbrains.com/issue/KT-37316
-expect class ArrayIndexOutOfBoundsException(message: String?) : IndexOutOfBoundsException
-
-internal expect inline fun <R> synchronized(lock: Any, block: () -> R): R
-
-expect open class IOException(message: String?, cause: Throwable?) : Exception {
-  constructor(message: String? = null)
+public expect open class IOException(message: String?, cause: Throwable?) : Exception {
+  public constructor(message: String? = null)
 }
 
-expect class ProtocolException(message: String) : IOException
-
-expect open class EOFException(message: String? = null) : IOException
-
-expect class FileNotFoundException(message: String? = null) : IOException
+public expect open class EOFException(message: String? = null) : IOException
```

Comment:
Does it make sense to make `EOFException` public right from the start?

Reply:
We're declaring that some methods may throw `EOFException`, so I expect someone will try to catch it.